symphony-bdk-core/src/test/java/com/symphony/bdk/core/auth/impl/OboAuthenticatorCertImplTest.java

```diff
@@ -0,0 +1,101 @@
+package com.symphony.bdk.core.auth.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.symphony.bdk.core.api.invoker.ApiRuntimeException;
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.core.test.MockApiClient;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class OboAuthenticatorCertImplTest {
+
+  private static final String APP_AUTHENTICATE = "/sessionauth/v1/app/authenticate";
+  private static final String APP_AUTHENTICATE_OBO_USERNAME = "/sessionauth/v1/app/username/{username}/authenticate";
+  private static final String APP_AUTHENTICATE_OBO_USERID = "/sessionauth/v1/app/user/{uid}/authenticate";
+
+  private OboAuthenticatorCertImpl authenticator;
+  private MockApiClient mockApiClient;
+
+  @BeforeEach
+  void init() {
+    this.mockApiClient = new MockApiClient();
+    this.authenticator = new OboAuthenticatorCertImpl(
+        "appId",
+        this.mockApiClient.getApiClient("/sessionauth")
+    );
+  }
+
+  @Test
+  void testAuthenticateByUsername() throws AuthUnauthorizedException {
+    this.mockApiClient.onPost(APP_AUTHENTICATE, "{ \"token\": \"1234\", \"name\": \"sessionToken\" }");
+    this.mockApiClient.onPost(APP_AUTHENTICATE_OBO_USERNAME.replace("{username}", "username"), "{ \"sessionToken\": \"1234\" }");
+    final AuthSession session = this.authenticator.authenticateByUsername("username");
+    assertNotNull(session);
```

Comment:
IMHO the checks on session is not required as we do have asserts on session after (so they would fail with NPE)
I would also recommend using line breaks to split different phases of the test, i.e Given/When/Then (the setup, what is tested and the checks)